packages/nodes-base/nodes/ActiveCampaign/ActiveCampaign.node.ts

```diff
@@ -1184,20 +1184,32 @@ export class ActiveCampaign implements INodeType {
 					responseData = { success: true };
 				}
 
+				responseData = this.helpers.returnJsonArray(responseData);
+				responseData = this.helpers.constructExecutionMetaData({ item: i }, responseData);
+
 				if (Array.isArray(responseData)) {
-					returnData.push.apply(returnData, responseData as IDataObject[]);
+					returnData.push.apply(returnData, responseData);
 				} else {
-					returnData.push(responseData as IDataObject);
+					returnData.push(responseData);
 				}
 			} catch (error) {
 				if (this.continueOnFail()) {
-					returnData.push({ error: error.message });
+					const executionErrorData = this.helpers.returnJsonArray({
```

Comment:
When we call `returnJsonArray` we actually bundle the existing object with `{ error: something, $json: somethingElse, itemIndex: i }` into another. Will this work properly when errors happen? I believe we might be wrapping again and causing the output to become invalid.

Reply:
Correct, I have already spoken with Jan regarding this.
I need to go back and manually build errors instead of using the `returnJsonArray` helpers in these situations.

Reply:
@krynble Are we deeming this wrong? I will continue to look into it.

I found that it seems to already exist in that manner for some nodes.

```
if (this.continueOnFail()) {
    return [this.helpers.returnJsonArray({ error: error.message })];
}
```